core/arch/riscv/kernel/entry.S

```diff
@@ -188,6 +202,8 @@ UNWIND(	.cantunwind)
 	mv	s3, a0
 	STR	x0, THREAD_CORE_LOCAL_FLAGS(s3)
 
+	mv	a0, s1		/* s1 contains saved device tree address */
+	mv	a1, x0		/* unused */
```

Comment:
`x0` seems to hold some value. It should better be zeroed before being passed to `boot_init_primary_late()`. Unless it's fine as is as per _arch/riscv/kernel/boot.c_ implementation.

Reply:
`x0` is hardwired to zero, same as ARM's `WZR/XZR`.

`mv a1, x0` is equivalent to `a1 = 0`.

Reply:
ok, understood. thanks.